kivy/effects/scroll.py

```diff
@@ -106,6 +107,15 @@ def on_value(self, *args):
             self.reset(scroll_max)
         else:
             self.scroll = self.value
+    
+    def apply_distance(self, distance):
+        if abs(distance) < self.min_distance:
+            self.velocity = 0
+
+        if distance < 0 and self.value > self.max:
+            self.value += distance
+        elif distance > 0 and self.value < self.min:
+            self.value += distance
```

Comment:
It's kinda confusing because if you look this part alone, it looks wrong. And the following code looks correct:

```python
        if distance < 0 and self.value > self.min:
            self.value += distance
        elif distance > 0 and self.value < self.max:
            self.value += distance
```

But when I actually changed the code like that, ScrollEffect didn't work correctly.